tests/phpunit/CRM/Civicase/Service/CaseSalesOrderContributionTest.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+use Civi\Api4\CaseSalesOrder;
+use CRM_Civicase_Service_CaseSalesOrderContribution as SalesOrderService;
+
+/**
+ * Runs tests on CRM_Civicase_Service_CaseSalesOrderContribution Service tests.
+ *
+ * @group headless
+ */
+class CRM_Civicase_Service_CRM_Civicase_Service_CaseSalesOrderContributionsTest extends BaseHeadlessTest {
+  use Helpers_PriceFieldTrait;
+  use Helpers_CaseSalesOrderTrait;
+
+  /**
+   * Setup data before tests run.
+   */
+  public function setUp() {
+    $this->generatePriceField();
+  }
+
+  /**
```

Comment:
I think you will need some more test cases for the other paths in this service, e.g :  one for using INVOICE_PERCENT another for  INVOICE_REMAIN , one for  discounted_percentage > 0 and one when it is = 0